### PR TITLE
docs/omf-release-0.1

### PR DESCRIPTION
**Pull request by karcass**
Separate PR for documentation, so that you don't have to dig around in the [catch-up PR](https://github.com/openmobilityfoundation/mds-core/pull/249).

This PR should be merged before the catch-up PR so that it's included.

Edits and requests for additional documentation are welcome and encouraged.

### CODE REVIEW

**Comment by sarob** [comment]
Looks great so far. One likely change to discuss. Jascha would like to make the repo root readme files read mostly the same. If we can all agree to that, then the current mds-core readme.md will need to change to something like DevWorkstationSetup.md. I will push a PR on the spec readme today as reference and comment here for review.

**Comment by sarob** [comment]
Starting to review dev workstation setup (after moving it from the root readme) which is now part of this PR. I went back through the openstack infrastructure management and project creation guides as reference for running local and remote tests.

**Comment by sarob** [comment]
While working on DevWorkstationSetup.md, I have been following the steps. This is what I have found so far

- Some errors were logged as part of the mds-core bootstrap script. I can post these to the PR or somewhere else more helpful.
- I got all the way to the mds-core build section, where I get  `curl: (52) Empty reply from server` after running  'curl localhost/agency’.
- yarn.lock has a new reference for chalk 4.0.0
- package.json requires the node reference be updated from 12.14.1 to 13.12.0 before I could run git commit
- Should I be pushing these two files to the mds-core repo?

**Comment by sarob** [comment]
I figured out the bootstrap script install. Small tweak now part of the doc. Now working through a bootstrap script build problem.

File: ARCHITECTURE.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+# Languages
+This implementation is built in [TypeScript](https://www.typescriptlang.org) v3.8.  The first draft of the Agency service was built in vanilla EMCAScript 6, and types were added as the system grew.
+
+TypeScript has a large developer community, a comprehensive set of toolchains, wide support in cloud providers, and is reasonably performant.  JavaScript is the [most popular](https://octoverse.github.com/) language inside GitHub, with TypeScript 7th.
+
+The mds-core endpoints are implemented using the popular framework [express.js](https://expressjs.com/).
+
+# Authentication & Identity
+We authenticate and authorize access to APIs and public-facing websites using JSON Web Tokens (JWT). JWTs are issued by Auth0 using the OAuth 2.0 protocol. 
+## Auth0
```
Comment (janedotx) [inline ARCHITECTURE.md:17]: Is this only meant to be an explanation of how our implementation works? I think a sentence or two about whether we recommend using Auth0 and the pros of cons of other authentication alternatives might be nice too.

File: ARCHITECTURE.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+Services in a service-oriented-architecture can communicate with each other via the same types of interfaces used by the outside clients (e.g. REST or SOAP), or they can use an internal message bus. We chose the latter for perceived wins in speed and efficiency. After considering knative eventing and kafka, we settled on [NATS](https://nats.io) as a more stable and low-operational-overhead solution. We have attempted to abstract away the message interface, however, such that we could later decide to plug in a different message bus.
+
+NATS is recognized as an incubating [CNCF](https://www.cncf.io) [open source project](https://github.com/nats-io) made available under the [Apache License](https://www.apache.org/licenses/LICENSE-2.0.html) and supporting a number of contemporary software languages.
+# Testing & Validation
+The repo currently has integration tests with code-coverage minimums on a per-project basis.  [TODO add more]
+# Versioning
+The services in mds-core will follow the MDS spec’s requirements for [API versioning](https://github.com/openmobilityfoundation/mobility-data-specification/blob/dev/general-information.md).
+# Shared Libraries
```
Comment (janedotx) [inline ARCHITECTURE.md:74]: This will need to be updated to add `mds-service-helpers` and `mds-webpack-config`

File: ARCHITECTURE.md (hunk 1 above)
Comment (wesleyfsmith) [reply ARCHITECTURE.md:17]: Yeah, I agree that adding some additional information about Auth0 would be great

File: ARCHITECTURE.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+## mds-providers
+Collection of information about supported micro-mobility providers.
+## mds-stream
+Collection of methods to send messages to NATS-Streaming. Currently supports sending vehicle events and telemetry.
+## mds-types
+Collection of TypeScript descriptions of types that are shared among components, including Agency’s notion of Device and VehicleEvent, event type enums for Provider and Agency, etc.
+## mds-utils
+Grab-bag of shared utility methods. Including, but not limited to: Programmatic representation of the MDS-Agency State Machine, date-time converters, and geospatial analysis methods.
+## mds-web-sockets
+Implementation of a basic websocket model for MDS. Enables real-time external consumption of data through a pub/sub model. See package documentation for specific examples.
```
Comment (wesleyfsmith) [inline ARCHITECTURE.md:96]: Should we mention that it's probably a better practice to use NATS instead of hooking into the web sockets directly?

**Comment by twelch** [comment]
@karcass Is this PR stale and can be closed?

**Comment by avatarneil** [comment]
Closing due to inactivity

**Pull request closed without merging**